README.md

```diff
@@ -188,38 +213,110 @@ We've recovered from the conversion error and kept going. Algebraic effects and
 handlers are similar to exceptions in that we can declare new effects:
 
 ```ocaml
-effect Conversion_failure : string -> int
+type _ Effect.t += Conversion_failure : string -> int Effect.t
 (* c.f. [exception Conversion_failure of string] *)
 ```
 
+Effects are declared by adding constructors to an [extensible variant type](https://kcsrk.info/webman/manual/extensiblevariants.html)
+defined in the `Effect` module.
+
 Unlike exceptions, performing an effect returns a value. The declaration here
 says that `Conversion_failure` is an algebraic effect that takes a string
 parameter, which when performed, returns an integer. 
 
 Just like exceptions, effects are values. The type of `Conversion_failure
-"MMXVII"` is `int effect`, where `int` is the result of performing the effect.
-We perform the effect with `perform : 'a eff -> 'a` primitive (c.f. `raise :
+"MMXVII"` is `int Effect.t`, where `int` is the result of performing the effect.
+We perform the effect with `perform : 'a Effect.t -> 'a` primitive (c.f. `raise :
 exn -> 'a (* bottom *)`). 
 
-Effect handlers are similar to exception handlers:
+Effect handlers are defined in the modules `Effect.Deep` and `Effect.Shallow`.
+We'll discuss the differences between the two later. (TODO THOMAS)
```

Comment:
What about this TODO?

Reply:
Oh I wrote that section already, will delete the todo

Reply:
Done